app/app/dialog/[id]/page.tsx

```diff
@@ -1,200 +1,48 @@
-                  <div className={`text-xs text-right mt-1 ${
-                    isOutgoing
-                      ? 'text-gray-300'
-                      : 'text-muted-foreground'
-                    }`}>
-                    {formatDate(msg.date)}
-                  </div>
-                </div>
-              </div>
-            </div>
-          )
-        })}
-      </div>
-    </div>
-  )
-}
-
-
-export default function Page () {
+export default function BackupSelectionPage() {
+  const { id } = useParams<{ id: string }>()
+  const [{ backups }] = useBackups()
   const router = useRouter()
-  const [{ client }] = useTelegram()
-  const [{ restoredBackup }, { restoreBackup, fetchMoreMessages }] = useBackups()
-  const params = useParams<{ id: string }>()
   const searchParams = useSearchParams()
-  const backupCid = searchParams.get('backupData')
-  const [userId, setUserId] = useState<string>()
-  
-  useEffect(() => {
-    const fetchBackup = async () => {
-      try {
-        if (!client.connected) await client.connect()
-          
-        const userId = (await client.getMe()).id.toString()
-        setUserId(userId)
-
-        await restoreBackup(backupCid!, params.id, 50)
-      } catch (error) {
-        console.error('Error in useEffect:', error)
-        
-      }
-    }
+  const type = searchParams.get('type') 
 
-    fetchBackup()
-  }, [client, backupCid, params.id, restoreBackup])
+  const dialogBackups = backups.items.filter((b) => b.params.dialogs.includes(id))
 
-  const handleFetchMoreMessages = async () => {
-    return fetchMoreMessages(restoredBackup.item?.messages.length || 0, 50)
+  const handleBackupClick = (e: React.MouseEvent, backupCid: string) => {
+    e.preventDefault()
+    router.push(`/dialog/${id}/backup/${backupCid}?type=${type}`)
   }
 
   return (
-    <Layouts isSinglePage isBackgroundBlue>
-       {restoredBackup.error && (
-        <ConnectError
-          open={!!restoredBackup.error}
-          error={restoredBackup.error}
-          onDismiss={() => router.back()}
-        />
-      )}
-      {restoredBackup.loading && <p className="text-center">Loading...</p>}
-      {!restoredBackup.loading && !restoredBackup.item && (
-        <div className="flex flex-col items-center justify-center h-screen">
-          <p className="text-lg font-semibold text-red-500">Dialog not found</p>
+    <Layouts isSinglePage>
+        <div className="flex flex-col gap-5 min-h-screen">
+        <h1 className="px-5">Select a Backup</h1>
+        <div className="flex flex-col">
+            {dialogBackups.map((backup) => (
+            <div
+                key={backup.data.toString()}
+                className="flex justify-between items-center px-5 py-3 border-b border-border cursor-pointer hover:bg-muted"
+                onClick={(e) => handleBackupClick(e, backup.data.toString())}
+            >
+                <div>
+                <p className="text-sm text-foreground/60">Backup Date:</p>
```

Comment:
This is a little repetitive. Might be nice to put "1 chat" or "3 chats, 7 pictures" or something?